packages/signed-cid/src/eip712Codec.ts

```diff
@@ -0,0 +1,82 @@
+import * as web3 from "web3"
+import { getBytes } from "ethers"
+import { TypedDataEncoder } from "ethers/hash"
+
+import { Action, Message, Session } from "@canvas-js/interfaces"
+type Codec = { name: string; code: number; encode: (value: any) => Iterable<Uint8Array> }
+
+function dynamicAbiEncodeArgs(args: Record<string, any>): string {
+	const types: string[] = []
+	const values: any[] = []
+	for (const key in args) {
+		types.push("string")
+		values.push(key)
+		if (typeof args[key] === "string") {
+			if (args[key].match(/^0x[0-9a-fA-F]*$/)) {
```

Comment:
this should match exactly 40 chars for addresses - it's possible someone will create an action with a txid or other 0x hexstring.

i wonder if for consistency we should just encode any 0x... hexstring as `bytes`, and specifically those of length 40 as `address`, which eliminates one source of ambiguity. it's still a potential footgun but much less of a surprising one in that case